assets/js/carouselClass.js

```diff
@@ -0,0 +1,105 @@
+export class CarouselClass {
+  constructor(container) {
+    // Created variables
+    this.container = container;
+    this.carouselList = this.container.querySelector(".carousel-list"),
+    this.carouselItems = this.container.querySelectorAll(".carousel-list__item"),
+    this.carouselWrap = this.container.querySelector(".carousel-wrap"),
+    this.btnNext = this.container.querySelector(".carousel-next"),
+    this.btnPrev = this.container.querySelector(".carousel-prev"),
+    this.pagination = this.container.querySelectorAll(".pagination-bullet");
+
+    // Calculated variables
+    this.carouselWrapWidth = this.carouselWrap.offsetWidth,
+    this.carouselItemsLength = this.carouselItems.length,
+    this.carouselItemsActiveIndex = 0,
+    this.isResizing = false;
+
+
+  }
+
+  // Add or disable switch transition
```

Comment:
Consider adding comments only when the logic is complex. Comments like these are unnecessary since method name itself already describes what it does.